### DISCUSSION TITLE
Assuming ownership of manifest update

### DISCUSSION THREAD

**Discussion by hculea** [ScoopInstaller/Main#5099] [General]
Hello there! 👋

I am a developer with the CLI team at 1Password. My team and I are looking into helping keep 1Password CLI up to date on Scoop. We'd like to build an automation updating the 1Password CLI Scoop manifest with every CLI release.

In this way, we want to ensure that the CLI version that Scoop points to is always the latest one, so we'd be able to reliably recommend Scoop as an installation method in our [Developer Documentation](https://developer.1password.com/docs/cli/get-started/).

To my question:

[This guide](https://github.com/ScoopInstaller/Scoop/wiki/App-Manifest-Autoupdate) was very helpful to get the latest version of [the manifest](https://github.com/ScoopInstaller/Main/blob/master/bucket/1password-cli.json) locally. However, I am not clear about how would this change be reflected in the Main bucket.

The wiki suggests a flow like:
```
.\bin\checkver -App ./path/to/local/cli/manifest -u
git commit -m "Updated apps"
git push
``` 
but I am left wondering if these are indeed all the steps, and whether these can be performed by a party without write permissions to the repository.

The [CONTRIBUTING.md](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md#for-scoop-buckets) guide suggests that contributions to the buckets should be made through a PR from a fork, would this be the process to follow for updating a manifest too? 

Am I missing anything?

Would love to find the best way to proceed here! Thank you!

Horia

**Comment by chawyehsu** [comment]
Glad to see you and your team decide to leverage Scoop to distribute software.

Since [1password-cli](https://github.com/ScoopInstaller/Main/blob/master/bucket/1password-cli.json) has already been added to the repository, technically you don't need to do anything to make it always be the latest one because we have a [workflow](https://github.com/ScoopInstaller/Main/blob/master/.github/workflows/excavator.yml) that runs [every 4 hours](https://github.com/ScoopInstaller/Main/actions/workflows/excavator.yml) to check package updates, and it will [update the manifest automatically](https://github.com/ScoopInstaller/Main/commits/master/bucket/1password-cli.json) when the upstream delivers new version of the software. But this relies on a stable distribution scheme - means the upstream should provide a consistent/regular download url for the software and a reliable url for checking the releases history, since the bot [relies on the checking strategy defined in the manifest](https://github.com/ScoopInstaller/Main/blob/master/bucket/1password-cli.json#L25-L38) to work will only replace the [version number and checksum](https://github.com/ScoopInstaller/Main/commit/8719c21a98c9409466c1522cb4cd8792cab98d10) of the package, and if the upsteram decides to update the distribution scheme, for instance, changing the download endpoint or package file name/structure, then they have to modify the manifest, submit a PR to reflect the changes.

There is one suggestion I would make is if [the upstream](https://app-updates.agilebits.com/product_history/CLI2) could provide checksums of the packages, so the bot doesn't need to calculate them on the fly and can grab them from the upstream directly.

**Reply by hculea** [reply]
Oh that's absolutely awesome! 

I'll raise up the checksum issue with my team as a possible future improvement 😄 

Even before then, Scoop [is now mentioned](https://developer.1password.com/docs/cli/get-started/#step-1-install-1password-cli) as an installation method for the 1Password CLI on Windows. 🎉 

Thank you for your detailed answer!

Horia
